Feature/#376 color variable naming structure

jomurgel opened the pull request:
Closes #376

### DESCRIPTION ###
- Removed color assignments from `_color-vars` formerly `_colors` in "variables" into `_color-assignments` partial to keep all additional global color variables separate.
- Moves all color variables into `_color-assignments` partial and ensures that not color variables eg. `$color-white` is used anywhere outside of the `_color-vars` or `_color-assignments` partials.
- Renamed partials for variables `prefix-vars.scss` to avoid having any scss partials with the same name to avoid confusion.
- Moved `_sprits.scss` into mixins.
- Added a host of new color variables for forms, inputs, buttons, etc. A new variable anywhere a color variable was used outside of `_color-vars` formerly `_colors` inside variables.
- Adds additional documentation and `sassdocs` support.
- General cleanup and linting.

jomurgel committed:
Adds docs to style index

jomurgel committed:
renames colors to color classes for mixin partial

jomurgel committed:
cleans up and adds doc to base index partial

jomurgel committed:
removes colors partial — now in index

jomurgel committed:
adds doc and alphabetizes

jomurgel committed:
adds partial docBlocks and cleans up

jomurgel committed:
rename variable partials

jomurgel committed:
cleans up transition variables

jomurgel committed:
updates focus variables

jomurgel committed:
moves form var to color assignments

jomurgel committed:
replaces px with rem

jomurgel committed:
replaces unitless line heigth with lh() mixin

jomurgel committed:
adds color assignments import and removes form-var

jomurgel committed:
moves elements & skinning to color assignments

jomurgel committed:
updates color map to use variables

jomurgel committed:
cleanup

jomurgel committed:
adds color assignment partials and consolodates assigments

jomurgel committed:
replaces color var with table var

jomurgel committed:
adds table variable

jomurgel committed:
updates variable name

jomurgel committed:
adds and updates modal vars

jomurgel committed:
adds comment vars and updates partial

jomurgel committed:
cleanup

jomurgel committed:
adds vars from assignments

jomurgel committed:
adds gutenberger block variables

jomurgel committed:
adds mobile navigation variable assignments

jomurgel committed:
adds additional navigation variable assignments

jomurgel committed:
updates hero overlay color

jomurgel committed:
adds pre color assignments

jomurgel committed:
moves sprites from base to mixins

jomurgel committed:
adds scaffolding partial var assignments

jomurgel committed:
replaces box shadow with global variable

jomurgel committed:
builds styles

jomurgel committed:
updates docfile

jomurgel committed:
disable eslint in gulpfile

jomurgel committed:
removes whitespace

jomurgel committed:
updates variable hex color — all lowercase

mharis commented:
@jomurgel My suggestion would be to keep color variables and assignments in the same file so that devs can stay in one file, easier to navigate and remember. This is just me being lazy.

gregrickaby reviewed (approved):
I think there's some truth to @mharis's suggestion. I'll leave that up to the group though.

This is amazing, @jomurgel! 🌮 👍

gregrickaby reviewed (dismissed):
@jomurgel Can you merge in `master` and recompile? Thanks!

coreymcollins commented:
I think I tend to agree with keeping the color variables and assignments all in one file. The less I have to poke around to find the place where a color is set, the better. The partial will wind up being long, but I think it would be nicer to have all of that stuff in one place.

Awesome work on this, Jo. This was a lot to take on and it looks good so far! Going to wait to run through a full review until after the merge conflict items are taken care of, but so far this is amazing.

Thanks!

jomurgel committed:
Merge branch 'master' into feature/#376-color-variable-naming-structure

jomurgel committed:
consolodate color assignments and color var — reorganizes and updates TOC

jomurgel committed:
removes color assignments and renames color var

jomurgel committed:
builds

jomurgel committed:
re-aligns variables

jomurgel commented:
@mharis @coreymcollins I consolidated the `_color-assignments.scss` and `_color-var.scss` into one partial, updated the TOC on that part and did some general cleanup.

Passing it back for another review 👍

coreymcollins commented on assets/sass/base/_scaffolding.scss:
```diff
@@ -18,7 +18,7 @@
 	// The section <header>
 	&-header {
-		border-bottom: rem(1) solid $color-alto;
+		border-bottom: rem(1) solid $color-scaffolding-border;
```
Borders should use `px` instead of `rem()`

coreymcollins commented on assets/sass/base/_scaffolding.scss:
```diff
@@ -48,16 +48,16 @@
-			border: rem(1) solid $color-alto;
-			color: $color-silver-chalice;
+			background-color: $color-pre-background;
+			border: rem(1) solid $color-pre-border;
```
Borders should use `px` instead of `rem()`

coreymcollins commented on assets/sass/modules/_content.scss:
```diff
@@ -71,7 +71,9 @@
+//-----------------------------------------
 // Asides
+//-----------------------------------------
 .blog .format-aside .entry-title,
```
I know this is already in the theme, but I think it might be best to remove this and instead add the `screen-reader-text` class and ARIA labels to the entry titles of posts where we want to hide the title. Right now, the post title is hidden so on an archive page I don't have any context about the post when the screen reader gets to it.

But maybe that's intentional? I'm not sure why these were hidden in the first place. 🤔

jomurgel replied:
That's a good question. I'm going to ignore this comment in this branch and open a new issue #429

coreymcollins commented on assets/sass/modules/navigation/_mobile-navigation.scss:
```diff
@@ -26,11 +26,11 @@
 
 		&:focus {
-			outline: rem(2) solid $color-white;
+			outline: rem(2) solid $color-mobile-menu-link;
```
We're setting `outline` with `rem()` here and then `px` later on in the file. We don't have a specification in our Units of Measurement graphic... do we need an update to a new version after over a year at v0.3.0?! 😮

coreymcollins commented on assets/sass/utilities/variables/_color-var.scss:
```diff
@@ -0,0 +1,316 @@
+//--------------------------------------------------------------
```
Some general comments here, rather than noting in a handful of comments on individual lines.

Sometimes the color variables are set like `$color-background-element` and other times `$color-element-background`. Sometimes variables use `color` twice, like `$color-element-color`.

I wonder if we can make this more consistent so everything is always `$color-background-element` for backgrounds and `$color-element` for colors?

Some examples from in this file:
`$color-background-screen-reader`
`$color-text-screen-reader`

I like those!

Then we have some others like this:
`$color-icon-close-background`
`$color-modal-close-color`

Those feel a little clunkier to me, especially the second one where `color` is used twice.

More than happy to compromise here if these were done a specific way for specific reasons, but just something I noticed that didn't seem to be done consistently.

jomurgel replied:
Good call. Should keep these consistent for sure.

coreymcollins commented on assets/sass/utilities/variables/_modal-var.scss:
```diff
@@ -13,7 +13,7 @@
 
 /// Define thin border radius.
-$border-radius-thin: 1px;
+$border-radius-thin: rem(1);
```
Border radius should be `px` according to our UoM graphic. Do we need to update the theme or update the graphic?

coreymcollins commented on assets/sass/utilities/variables/_transition-var.scss:
```diff
@@ -0,0 +1,30 @@
+$transition-transform: transform  $speed $ease;
+
+/// Transform Hamburger
+$transition-hamburger: transform $speed, background-color $speed $ease;
```
Can we call a transition variable from a transition variable? For instance, instead of doing what we have now do `$transition-hamburger: $transition-transform, $transition-bg-color;`?

Since we're not customizing the speed or easing, I'm wondering if that could work without Gulp getting weird about it.

jomurgel replied:
This would work, yeah, as long as the one being called is defined before the other. ie, `$transition-transform` is defined before `$transition-bg-color`

coreymcollins reviewed (changes requested):
The amount of work here is staggering. You rule for taking this on!

I have some comments on various partials here and there. Nothing too crazy – some questions/thoughts in some partials and some small unit of measurement tweaks to be made.

I think having these variables all in one partial is going to be awesome, and I love how many different variables are set for colors now with one place to find and manage them. Awesome work, Jo!

jomurgel committed:
updates transition partial — adds names to variables for reusability

jomurgel committed:
changes rems to px

jomurgel committed:
updates rem to px

jomurgel committed:
updates color variables to use `$color-type-element` or `$color-element` format.

jomurgel committed:
builds

jomurgel committed:
Merge branch 'master' into feature/#376-color-variable-naming-structure

jomurgel committed:
move speed into all transition variables

jomurgel committed:
builds

jomurgel commented:
@coreymcollins I made several updates.

- Most notably converting all of the color variables to the `$color-type-element` or `$color-element` format throughout. 
- Opened up #429 to talk about those SRT classes and ARIA labels.
- Updated rem to px in a few locations.
- Updated the transitions partial to utilize variables for the transition names (with the exception of all which doesn't need to be reused).

coreymcollins committed:
Adds $ease to a couple of missing spots in the transition variables file

coreymcollins commented:
Awesome work here, Jo!

I pushed up one commit here:
https://github.com/WebDevStudios/wd_s/pull/382/commits/0e03ba4836a745f5fa0c8d0093ddc58b1d1a8f76

A couple of the variables set for transitions were missing the `$ease` variable when multiple transitions were being used.

Other than that, this looks really great.

I'd love to have a second pair of eyes give this the OK as well since there is a *_lot_* to look at.

Once it's approved by two of us, `master` will need to be merged in to resolve any conflicts there as we've had a pretty active day of merging things in so I'm sure there will be some conflicts.

Thanks again for your hard work on this!

jomurgel committed:
Merge branch 'master' into feature/#376-color-variable-naming-structure

jomurgel committed:
adds missing variable mixin

jomurgel committed:
replaces duplicate $transition-all with $transition-height.

jomurgel committed:
replaces multiple transitions for accordion with one.

jomurgel committed:
updates variable name for icon background color

jomurgel committed:
builds scss

jomurgel commented:
@gregrickaby @coreymcollins I've merged in `master` and recompiled all the things. In doing so I also fixed a couple transition variables, added a transition variable for the accordion and fixed a couple of variable names that had since been updated.

mharis commented:
I did a review @coreymcollins and this seems good overall.

One note would that maybe we can move
```
/// Global input padding.
$padding-input: rem(6);
``` 

From color variables file to the `assets/sass/modules/_forms.scss` file.

jomurgel committed:
adds button padding

jomurgel committed:
install package

jomurgel committed:
removes padding from color var partial

jomurgel committed:
builds

jomurgel commented:
I ended up removing this variable altogether and hardcoding the value in the `_forms.scss`, `_gravity-forms.scss`, and `_buttons.scss` respectively.

Agreed that that should be moved from the `_color-var.scss` partial as it's not a color var, but since these elements are always tweaked independently of one-another I don't think it makes much sense to have them using the same variable.

jomurgel commented:
Alrighty. I ran through this PR again. I think it's as clean and purposeful as it can be until we actually give some purposeful design elements to the default WD_s. Just need another pair of eyes from @coreymcollins, @gregrickaby, and @ImBigWill.

jomurgel committed:
audit npm packages

jomurgel committed:
builds

jomurgel committed:
installs fixes dependencies

jomurgel committed:
brings package.json current with master

jomurgel committed:
fixes json parse issue

jomurgel committed:
builds

jomurgel committed:
rebuilds all

jomurgel committed:
adds gulp replace to remove spaces from project.js

jomurgel committed:
rebuilds all

jomurgel committed:
builds all

jomurgel committed:
rebuilds all

jomurgel commented:
@gregrickaby I've merged in and fixed issue getting through buddy. Should be gtg here. 👍

coreymcollins committed:
Fixes a missing semicolon error in comments partial

coreymcollins committed:
Ignores a class-name rule for SassLint

coreymcollins commented:
I ran a final Sass Lint against this branch and found only two issues, for which I have pushed up fixes
https://github.com/WebDevStudios/wd_s/pull/382/commits/6483eea01a4f592ea8793bedfea836c748c8c687:
- https://github.com/WebDevStudios/wd_s/pull/382/commits/4307679a26bdf042844bcb9e7ee7ef6ef4d57f7e
- https://github.com/WebDevStudios/wd_s/pull/382/commits/6483eea01a4f592ea8793bedfea836c748c8c687

Greg's main request previously was to resolve merge conflicts, which was done, and Haris, Will, and myself have reviewed the branch. I think we're looking good here!

Going to dismiss Greg's lingering review (as he's hanging out with Donald Duck right now, the lucky son of a...) and merge this puppy in.

Thanks!
